instagram-app/src/components/CommentSection/CommentSection.js

```diff
@@ -0,0 +1,121 @@
+import React from "../../../node_modules/react";
+import ls from "local-storage";
+import PropTypes from "../../../node_modules/prop-types";
+import moment from "../../../node_modules/moment/moment";
+import Comment from "./Comment";
+import NewCommentBox from "./NewCommentBox";
+import styled from "styled-components"
+
+// BEGIN STYLING
+
+const SCCommentSectionDiv = styled.div`
+  width: 96%;
+  margin: 0 2%;
+`
+const SCTimestamp = styled.div`
+  font-size: .8rem;
+  color: grey;
+  margin-bottom: 10px;
+`
+
+// END STYLING
+
+class CommentSection extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      newInput: "",
+      newInputHeight: 18,
+      comments: this.props.comments,
+      // timestamp: this.props.timestamp
+    };
+  }
+
+  componentDidMount() {
+    this.setState({
+      comments: ls.get(`comments_${this.props.id}`) || this.state.comments,
+      // timestamp: ls.get(`last-comment_${this.props.id}`) || this.state.timestamp
+    });
+  }
+
+  handleInputChanges = ev => {
+    this.setState({
+      newInput: ev.target.value,
+      newInputHeight: ev.target.scrollHeight,
+    });
+  };
+
+  addNewComment = ev => {
+    ev.preventDefault();
+    if (this.state.newInput === "") {
+      return;
+    }
+    const currentUser = this.props.currentUser;
+    const text = this.state.newInput;
+    // const freshTimestamp = moment().format("MMMM Do YYYY, hh:mm:ss a");
+    ls.set(`comments_${this.props.id}`, [
+      ...this.state.comments,
+      { username: currentUser, text: text }
+    ]);
+    // ls.set(`last-comment_${this.props.id}`, freshTimestamp);
+    this.setState({
+      newInput: "",
+      comments: [
+        ...this.state.comments,
+        { username: currentUser, text: text }
```

Comment:
This works for our app, but I hope it makes sense that we don't want to keep and update comments state here. We want to pass a method from PostPage and update our data at our `single source of truth`